hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java

```diff
@@ -70,6 +71,7 @@ public class KeyValueContainerCheck {
   private String metadataPath;
   private HddsVolume volume;
   private KeyValueContainer container;
+  private DirectBufferPool bufferPool = new DirectBufferPool();
```

Comment:
The KeyValueContainerCheck is instantiated when background scanner scans a container. So it's going to create thousands of DirectBufferPool instances.

Using DirectBufferPool is a good idea. However you need to move it one layer up. Or make it a static object, like this one: https://github.com/apache/hadoop/blob/3b416c036c0c55e6f4813bd7f29241c94357c22e/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/client/impl/BlockReaderLocal.java#L68

Reply:
done